web/src/components/Footer.tsx

```diff
@@ -0,0 +1,106 @@
+import {
+  Box,
+  Flex,
+  IconButton,
+  Text,
+  useColorMode,
+  useColorModeValue,
+} from "@chakra-ui/react";
+import { BsLinkedin } from "react-icons/bs";
+import { VscGithub, VscGithubInverted, VscTwitter } from "react-icons/vsc";
+import { MarkdownText } from "./MarkdownText";
+
+export const Footer = () => {
+  const { colorMode } = useColorMode();
+  return (
+    <>
+      <Box
+        bg={useColorModeValue("#ECE8FD", "#2F206E")}
+        justifyContent={"center"}
+        alignItems={"center"}
+        zIndex={5}
+        fontSize={"0.85em"}
+        padding={"10px"}
+      >
+        <Flex
+          direction={"column"}
+          justifyContent={"center"}
+          gap={1}
+          alignItems={"center"}
+        >
+          <MarkdownText>
+            RealTime Digital Marketing is a demo application running on
+            [**SingleStoreDB**](https://singlestore.com).
+          </MarkdownText>
+
+          <Flex
+            direction={"row"}
+            gap={8}
+            fontWeight={"bold"}
+            fontSize={"0.85em"}
+          >
+            <Flex
+              direction={"row"}
+              gap={1}
+              justifyContent={"center"}
+              alignItems={"center"}
+            >
+              <Text>Repository</Text>
+              <IconButton
+                aria-label="Github Repo"
+                size="sm"
+                background={colorMode === "light" ? undefined : "black"}
```

Comment:
according to our code style guide, please use if/else blocks https://memsql.atlassian.net/wiki/spaces/MCDB/pages/1185873978/Code+Style+Best+Practices+Guide#Use-if-condition-instead-of-%26%26-or-%3F-%3A

Reply:
Using if else block here will be bit tricky. the reason of if else as per docs share is for easy understanding and avoid scratching head

But the way to use if else for component argument I think is more tricker here



you can say like

background={if (cond) return else return 2}

you have to wrap if inside function plus you have to make a call like this



background={(() => if (cond) return else return 1})()

notice the round brackets they are more difficult to  trace

should I still go for if else ??

Reply:
In this case, I think it makes it easier to read if you use if/else but extract things to variables.
```
let backgroundColor;
let githubIcon = <VscGithub size="1.2em" />;

if (colorMode === "dark") {
  backgroundColor = "black";
  githubIcon = <VscGithubInverted size="1.2em" />;
}
```
this way you just need to run one comparison and you can use these variables:
```
<IconButton
  aria-label="Github Repo"
  size="sm"
  background={backgroundColor}
  icon={githubIcon}
```

Reply:
this whole code was restructured to more generalised Icon Button should not have this concern now